sycamore/evaluation/pipeline.py

```diff
@@ -1,116 +1,143 @@
-                            {
-                                "hybrid": {
-                                    "queries": [
-                                        {"match": {"text_representation": doc["question"]}},
-                                        {
-                                            "neural": {
-                                                "embedding": {
-                                                    "query_text": doc["question"],
-                                                    "model_id": self._os_config["embedding_model_id"],
-                                                    "k": self._os_config["neural_search_k"],
-                                                }
-                                            }
-                                        },
-                                    ]
+        query["query"] = {
+            "_source": {"excludes": ["embedding"]},
+            "query": {
+                "hybrid": {
+                    "queries": [
+                        {"match": {"text_representation": doc["question"]}},
+                        {
+                            "neural": {
+                                "embedding": {
+                                    "query_text": doc["question"],
+                                    "model_id": self._os_config["embedding_model_id"],
+                                    "k": self._os_config["neural_search_k"],
+                                    "filter": [{"exists": {"field": "parent_id"}}],
                                 }
                             }
-                        ],
-                        "filter": [{"exists": {"field": "parent_id"}}],
-                    }
+                        },
+                    ]
+                }
+            },
+            "size": self._os_config["size"],
+        }
+
+        if "llm" in self._os_config:
+            query["params"] = {"search_pipeline": self._os_config["search_pipeline"]}
+            query["query"]["ext"] = {
+                "generative_qa_parameters": {
+                    "llm_question": doc["question"],
+                    "context_size": self._os_config["context_window"],
+                    "llm_model": self._os_config["llm"],
                 },
-                "size": self._os_config["size"],
+                "rerank": {"query_context": {"query_text": doc["question"]}},
             }
+        if "filters" in doc:
+            query["query"] = self._add_filter(query["query"], doc["filters"])
         return query
 
-    def _evaluate_queries(self, query_result: Document) -> Document:
+    def _process_queries(self, query_result: Document) -> Document:
+        logger.debug(("Query result: " + str(query_result.keys())))
         assert query_result.type == "OpenSearchQueryResult"
         result = EvaluationDataPoint(query_result)
+        result.generated_source_documents = [Element(hit) for hit in query_result["hits"]]
```

Comment:
think this need to be `... hit in query_results['hits']['hits']`?

Reply:
The term might be a little confusing, this is the internal query_result object of type (OpenSearchQueryResult). That has a hit property that is populated when parsing the raw os result in OpenSearchQueryExecutor